curve-lp/contracts/StrategyCurveAave.sol

```diff
@@ -0,0 +1,198 @@
+    uint256 public optimal = 1; // this is the token we sell into, 0 DAI, 1 USDC, 2 USDT, 3 WBTC, 4 WETH
+
+    IERC20 internal constant dai = IERC20(0xd586E7F844cEa2F87f50152665BCbc2C279D8d70);
+    IERC20 internal constant usdc = IERC20(0xA7D7079b0FEaD91F3e65f86E8915Cb59c1a4C664);
+    IERC20 internal constant usdt = IERC20(0xc7198437980c041c805A1EDcbA50c1Ce5db95118);
+
+    IUniswapV2Router02 internal mainRouter = IUniswapV2Router02(0x60aE616a2155Ee3d9A68541Ba4544862310933d4); // this is the router we swap with except for CRV
+    IUniswapV2Router02 internal crvRouter = IUniswapV2Router02(0x60aE616a2155Ee3d9A68541Ba4544862310933d4); // this is the router we swap CRV with, Sushi
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor(address _vault, address _gauge, string memory _name) public StrategyCurveBase(_vault, _gauge) {
+        // You can set these parameters on deployment to whatever you want
+        maxReportDelay = 1 days; // 1 day in seconds
+        healthCheck = 0x6fD0f710f30d4dC72840aE4e263c22d3a9885D3B;
+
+        // these are our standard approvals. want = Curve LP token
+        want.approve(address(_gauge), type(uint256).max);
+        crv.approve(address(crvRouter), type(uint256).max);
+        wavax.approve(address(mainRouter), type(uint256).max);
+
+        // set our strategy's name
+        stratName = _name;
+
+        // these are our approvals and path specific to this contract
+        dai.approve(address(curve), type(uint256).max);
+        usdc.approve(address(curve), type(uint256).max);
+        usdt.safeApprove(address(curve), type(uint256).max);
+
+        // start off with usdc
+        targetToken = address(usdc);
+    }
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+    // these will likely change across different wants.
+
+    function prepareReturn(uint256 _debtOutstanding)
+    internal
+    override
+    returns (
+        uint256 _profit,
+        uint256 _loss,
+        uint256 _debtPayment
+    )
+    {
+        // harvest our rewards from the gauge
+        gauge.claim_rewards();
+        uint256 crvBalance = crv.balanceOf(address(this));
+        uint256 wavaxBalance = wavax.balanceOf(address(this));
+
+        // if we claimed any CRV, then sell it
+        if (crvBalance > 0 && voter != address(0)) {
+            // keep some of our CRV to increase our boost
+            uint256 sendToVoter = crvBalance.mul(keepCRV).div(FEE_DENOMINATOR);
+            if (keepCRV > 0) {
+                crv.safeTransfer(voter, sendToVoter);
+            }
+
+            // check our balance again after transferring some crv to our voter
+            crvBalance = crv.balanceOf(address(this));
+        }
+
+        // sell our WAVAX and CRV if we have any
+        if (wavaxBalance > 0 || crvBalance > 0) {
+            _sell(wavaxBalance, crvBalance);
+        }
+
+        // deposit our balance to Curve if we have any
+        if (optimal == 0) {
```

Comment:
is it possible to set the same logic as minichef, and remove the if as Val suggest ? in case we switch optimal without harvesting , we deposit the dust found if available to the pool